### ISSUE TITLE
selectors with moz prefix first are removed when used with autoprefixer

### ISSUE THREAD

**Issue by lukekarrys** [postcss/postcss#506]
This happens with the latest `postcss` (5.0.1) and `autoprefixer-core` (5.2.1). It only happens if the autoprefixer plugin is used, but I cloned the autoprefixer repo as well and experimented to see if I could get a failing test, but there was already a test for this, and I coudn't get it to fail. Also if I downgrade to postcss@4.1.16 then the problem goes away.

I know that `-moz-` prefixes aren't needed when using animations, but I was trying to consume some old 3rd party CSS that used them (but still used the unprefixed version) and the whole rule ended up getting emptied so the animation wouldn't work in any browser.

Here's a test case that I created:

```js
function toCss (rules) {
  return '.selector {\n  ' + rules.join('\n  ') + '\n}';
}

function process () {
  var css = toCss(Array.prototype.slice.call(arguments));
    postcss([autoprefixer()]).process(css).then(function (result) {
        console.log('\nINPUT');
        console.log(css);
        console.log('\nOUTPUT');
        console.log(result.css || 'EMPTY');
        console.log('---------------------');
    });
}

var rule = 'animation: loader;';
// No prefix, works
process(rule);
// Webkit only, works
process('-webkit-' + rule, rule);
// Webkit, then moz, works
process('-webkit-' + rule, '-moz-' + rule, rule);
// Moz, then webkit, does not work
process('-moz-' + rule, '-webkit-' + rule, rule);
// Moz only, does not work
process('-moz-' + rule, rule);
```

And here is the output from running that with `postcss@5.0.1`:
```css
INPUT
.selector {
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
          animation: loader;
}
---------------------
INPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}
---------------------
INPUT
.selector {
  -webkit-animation: loader;
  -moz-animation: loader;
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}
---------------------
INPUT
.selector {
  -moz-animation: loader;
  -webkit-animation: loader;
  animation: loader;
}

OUTPUT
EMPTY
---------------------
INPUT
.selector {
  -moz-animation: loader;
  animation: loader;
}

OUTPUT
EMPTY
---------------------

```

And the output from `postcss@4.1.16`:
```css
INPUT
.selector {
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
          animation: loader;
}
---------------------
INPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}
---------------------
INPUT
.selector {
  -webkit-animation: loader;
  -moz-animation: loader;
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}
---------------------
INPUT
.selector {
  -moz-animation: loader;
  -webkit-animation: loader;
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
  animation: loader;
}
---------------------
INPUT
.selector {
  -moz-animation: loader;
  animation: loader;
}

OUTPUT
.selector {
  -webkit-animation: loader;
          animation: loader;
}
---------------------
```

**Comment by ai** [comment]
The default Autoprefixer behaviour is to remove outdated prefixes, so by API it must remove `-moz-animation: loader`.

You can disable this behaviour by set `autoprefixer({ remove: false })` https://github.com/postcss/autoprefixer#outdated-prefixes

**Comment by lukekarrys** [comment]
But shouldn't something like

```css
.selector {
  -moz-animation: loader;
  animation: loader;
}
```

End up as

```css
.selector {
  animation: loader;
}
```

instead of removed altogether?

**Comment by ai** [comment]
Oops, sorry, seems like I miss some example.

Can you please post only broken examples? ^_^

**Comment by lukekarrys** [comment]
Sorry I didn't know if it would be helpful to post test cases that didnt fail as well. Here are the ones that fail. If I run them through postcss@5.0.1 with autoprefixer then `result.css` is empty. It works with postcss@4.1.16.

```css
.selector {
  -moz-animation: loader;
  -webkit-animation: loader;
  animation: loader;
}

.selector {
  -moz-animation: loader;
  animation: loader;
}
```

**Comment by ai** [comment]
It is really wierd O_o. I am too drunk to fix it today :(.

I think issue is somewhere in 4.x AST fix https://github.com/postcss/postcss/blob/master/lib/container.es6#L305

**Comment by ai** [comment]
Fixed 9fce511

**Comment by ai** [comment]
Released in 5.0.2.
